qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,600 @@
+        if val in slots:
+            raise InvalidValue("Slots cannot be exclusively closed.")
+        if val == "":
+            raise InvalidValue('An empty string may cause all channels and '
+                               'associated backplane relays to open. Use '
+                               '"open_channel" parameter instead.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels or channel '
+                               'ranges and associated backplane relays.')
+        self.write(f"channel.exclusiveclose('{val}')")
+
+    def _set_exclusive_slot_close(self, val: str) -> None:
+        slots = ['allslots', *self._get_slot_name()]
+        if val in slots:
+            raise InvalidValue("Slots cannot be exclusively closed.")
+        if val == "":
+            raise InvalidValue('Argument cannot be an empty string.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels or channel '
+                               'ranges and associated backplane relays.')
+        self.write(f"channel.exclusiveslotclose('{val}')")
+
+    def _get_channel_connect_rule(self) -> str:
+        connect_rule = {1: 'BREAK_BEFORE_MAKE',
+                        2: 'MAKE_BEFORE_BREAK',
+                        0: 'OFF'}
+        rule = self.ask('channel.connectrule')
+        return connect_rule[int(float(rule))]
+
+    def _set_channel_connect_rule(self, val: str) -> None:
+        self.write(f'channel.connectrule = channel.{val}')
+
+    def _get_gpib_status(self) -> str:
+        return self.ask('comm.gpib.enable')
+
+    def _set_gpib_status(self, val: Union[str, bool]) -> None:
+        self.write(f'comm.gpib.enable = {val}')
+
+    def _get_lan_status(self) -> str:
+        return self.ask('comm.lan.enable')
+
+    def _set_lan_status(self, val: Union[str, bool]) -> None:
+        self.write(f'comm.lan.enable = {val}')
+
+    def _get_gpib_address(self) -> int:
+        return int(float(self.ask('gpib.address')))
+
+    def _set_gpib_address(self, val: int) -> None:
+        self.write(f'gpib.address = {val}')
+
+    def get_closed_channels(self, val: str) -> str:
+        """
+        Queries for the closed channels.
+
+        Args:
+            val: A string representing the channels,
+                backplane relays or channel patterns to be queried.
+        """
+        if val == "":
+            raise InvalidValue('Argument cannot be an empty string.')
+        if not self._validator(val):
+            raise InvalidValue(f'{val} is not a valid specifier. '
+                               'The specifier should be channels, channel '
+                               'ranges, slots, backplane relays or "allslots".')
+        return self.ask(f"channel.getclose('{val}')")
```

Comment:
It might be worth parsing the 'nil' returned if no channels are closed to None - I can imagine this would be a nicer check in a routing situation so you could write something like:
if smatrix.get_closed_channels("slot1") is not None:
     foo